ui/components/multichain/account-list-item-menu/account-list-item-menu.js

```diff
@@ -82,116 +90,178 @@ export const AccountListItemMenu = ({
   };
 
   ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  const isCustodial = keyring?.type ? /Custody/u.test(keyring.type) : false;
   const accounts = useSelector(getMetaMaskAccountsOrdered);
-  const isCustodial = /Custody/u.test(keyring.type);
+
   const mmiActions = mmiActionsFactory();
   ///: END:ONLY_INCLUDE_IN
 
+  // Handle Tab key press for accessibility inside the popover and will close the popover on the last MenuItem
+  const lastItemRef = useRef(null);
+  const accountDetailsItemRef = useRef(null);
+  const removeAccountItemRef = useRef(null);
+  const removeJWTItemRef = useRef(null);
+
+  // Checks the MenuItems from the bottom to top to set lastItemRef on the last MenuItem that is not disabled
+  useEffect(() => {
+    if (removeJWTItemRef.current) {
+      lastItemRef.current = removeJWTItemRef.current;
+    } else if (removeAccountItemRef.current) {
+      lastItemRef.current = removeAccountItemRef.current;
+    } else {
+      lastItemRef.current = accountDetailsItemRef.current;
+    }
+  }, [
+    removeJWTItemRef.current,
+    removeAccountItemRef.current,
+    accountDetailsItemRef.current,
+  ]);
+
+  const handleKeyDown = (event) => {
+    if (event.key === 'Tab' && event.target === lastItemRef.current) {
+      // If Tab is pressed at the last item to close popover and focus to next element in DOM
+      onClose();
+    }
+  };
+
+  // Handle click outside of the popover to close it
+  const popoverDialogRef = useRef(null);
+
+  const handleClickOutside = (event) => {
+    if (
+      popoverDialogRef?.current &&
+      !popoverDialogRef.current.contains(event.target)
+    ) {
+      onClose();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
   return (
-    <Menu
-      anchorElement={anchorElement}
-      className="account-list-item-menu"
-      onHide={onClose}
+    <Popover
+      className="multichain-account-list-item-menu__popover"
+      referenceElement={anchorElement}
+      role={PopoverRole.Dialog}
+      position={PopoverPosition.Bottom}
```

Comment:
What do you think about using `BottomEnd`? So the menu aligns with the button?


https://github.com/MetaMask/metamask-extension/assets/8112138/ff2b0a03-bdbe-4b18-b43c-743cb0530b8f